From b985a91b4c42fefee7fe7c527c73af016a1ab88a Mon Sep 17 00:00:00 2001
From: "Wulf C. Krueger" <philantrop@exherbo.org>
Date: Sat, 6 Oct 2012 11:37:44 +0200
Subject: [PATCH] configure.ac: Add a --with-python option.

Being explicit about Python support rather than auto-detecting it and
acting upon the result, specifying it as an option gains us more
control about both dependencies and the resulting build.

Furthermore, relying purely on auto-detection can lead to problems for
source-based distros. E. g. systemd being built before *both* 32-bit &
64-bit ABIs are installed will lead to build failures as systemd's
build system will pick up either 32-/64-bit Python, conclude both are
available and fail if that's not the case.
---
 configure.ac | 31 +++++++++++++++++++------------
 1 file changed, 19 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5d7d0c2..6308270 100644
--- a/configure.ac
+++ b/configure.ac
@@ -90,19 +90,26 @@ fi
 # we use python to build the man page index, and for systemd-python
 have_python=no
 have_python_devel=no
-AM_PATH_PYTHON(,, [:])
-if test "$PYTHON" != : ; then
-        have_python=yes
-        AC_PATH_PROG([PYTHON_CONFIG], python-config)
-
-        if test -n "$PYTHON_CONFIG" ; then
-                have_python_devel=yes
-                PYTHON_CFLAGS="`$PYTHON_CONFIG --cflags`"
-                PYTHON_LIBS="`$PYTHON_CONFIG --libs`"
-                AC_SUBST(PYTHON_CFLAGS)
-                AC_SUBST(PYTHON_LIBS)
+
+AC_ARG_WITH([python],
+        [AS_HELP_STRING([--with-python], [Build the man page index and systemd-python (default: disabled)])])
+
+AS_IF([test "x$with_python" = "xyes"], [
+        AM_PATH_PYTHON(,, [:])
+        if test "$PYTHON" != : ; then
+                have_python=yes
+                AC_PATH_PROG([PYTHON_CONFIG], python-config)
+
+                if test -n "$PYTHON_CONFIG" ; then
+                        have_python_devel=yes
+                        PYTHON_CFLAGS="`$PYTHON_CONFIG --cflags`"
+                        PYTHON_LIBS="`$PYTHON_CONFIG --libs`"
+                        AC_SUBST(PYTHON_CFLAGS)
+                        AC_SUBST(PYTHON_LIBS)
+                fi
         fi
-fi
+])
+
 AM_CONDITIONAL([HAVE_PYTHON], [test "$have_python" = "yes"])
 AM_CONDITIONAL([HAVE_PYTHON_DEVEL], [test "$have_python_devel" = "yes"])
 
-- 
1.7.12.2

